src/cephadm/cephadm

```diff
@@ -1805,7 +1805,14 @@ def command_deploy():
         daemon_port = Monitoring.port_map[daemon_type]
         if port_in_use(daemon_port):
             raise Error("TCP Port '{}' required for {} is already in use".format(daemon_port, daemon_type))
-        elif args.image == DEFAULT_IMAGE:
+
+        # Monitoring metadata is nested dicts, so asking mypy to ignore
```

Comment:
couldn't we still use `Dict[Any, Any]` then?

Reply:
I'm not sure why mypy fails to use `Dict[Any, Any]` here, but it does:
`cephadm:1809: error: Incompatible types in assignment (expression has type "object", variable has type "Dict[Any, Any]")`
I understand now why `Any` was chosen as type declaration. Basically nothing else worked when I tested it. However, I was able to overcome that problem by annotating the initial dictionary (`Monitoring.components`) with `Dict[str, dict]`.